extensions/cornerstone-dicom-pmap/src/viewports/OHIFCornerstonePMAPViewport.tsx

```diff
@@ -0,0 +1,162 @@
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect, useRef } from 'react';
+import { useViewportGrid } from '@ohif/ui';
+import createPMAPToolGroupAndAddTools from '../utils/initPMAPToolGroup';
+
+const PMAP_TOOLGROUP_BASE_NAME = 'PMAPToolGroup';
+
+function OHIFCornerstonePMAPViewport(props: withAppTypes) {
+  const { displaySets, viewportOptions, displaySetOptions, servicesManager, extensionManager } =
+    props;
+  const viewportId = viewportOptions.viewportId;
+  const { displaySetService, toolGroupService, customizationService } = servicesManager.services;
+  const toolGroupId = `${PMAP_TOOLGROUP_BASE_NAME}-${viewportId}`;
+
+  // PMAP viewport will always have a single display set
+  if (displaySets.length !== 1) {
+    throw new Error('PMAP viewport must have a single display set');
+  }
+
+  const pmapDisplaySet = displaySets[0];
+  const [viewportGrid, viewportGridService] = useViewportGrid();
+  const referencedDisplaySetRef = useRef(null);
+  const { viewports, activeViewportId } = viewportGrid;
+  const referencedDisplaySet = pmapDisplaySet.getReferenceDisplaySet();
+  const referencedDisplaySetMetadata = _getReferencedDisplaySetMetadata(
+    referencedDisplaySet,
+    pmapDisplaySet
+  );
+
+  referencedDisplaySetRef.current = {
+    displaySet: referencedDisplaySet,
+    metadata: referencedDisplaySetMetadata,
+  };
+
+  const getCornerstoneViewport = useCallback(() => {
+    const { displaySet: referencedDisplaySet } = referencedDisplaySetRef.current;
+    const { component: Component } = extensionManager.getModuleEntry(
+      '@ohif/extension-cornerstone.viewportModule.cornerstone'
+    );
+
+    displaySetOptions.unshift({});
+    const [pmapDisplaySetOptions] = displaySetOptions;
+
+    // Make sure `options` exists
+    pmapDisplaySetOptions.options = pmapDisplaySetOptions.options ?? {};
+
+    Object.assign(pmapDisplaySetOptions.options, {
+      colormap: {
+        name: 'rainbow',
+        opacity: [
+          { value: 0, opacity: 0.5 },
+          { value: 1, opacity: 1 },
+        ],
+      },
+    });
+
+    return (
+      <Component
+        {...props}
+        displaySets={[referencedDisplaySet, pmapDisplaySet]}
```

Comment:
I think the order here is crucial. The reverse sequence would not be effective. Please add a comment explaining this.